src/api/cow/cow.spec.ts

```diff
@@ -0,0 +1,441 @@
+  expect(price?.amount).toEqual('1234567890')
+  expect(price?.token).toEqual('0x6810e776880c02933d47db1b9fc05908e5386b96')
+})
+
+test('Invalid: Get Price Quote (Legacy) with unexisting token', async () => {
+  fetchMock.mockResponseOnce(
+    JSON.stringify({
+      errorType: 'NotFound',
+      description: "You've passed an invalid URL",
+    }),
+    { status: HTTP_STATUS_NOT_FOUND }
+  )
+  try {
+    await cowSdk.cowApi.getPriceQuoteLegacy({
+      baseToken: 'unexistingToken',
+      quoteToken: 'unexistingToken',
+      amount: '1234567890',
+      kind: OrderKind.BUY,
+    } as PriceQuoteParams)
+  } catch (e) {
+    const error = e as QuoteError
+    expect(error.message).toEqual('Token pair selected has insufficient liquidity')
+    expect(fetchMock).toHaveBeenCalledTimes(1)
+    expect(fetchMock).toHaveBeenCalledWith(
+      'https://api.cow.fi/rinkeby/api/v1/markets/unexistingToken-unexistingToken/buy/1234567890',
+      {
+        body: undefined,
+        headers: {
+          'Content-Type': 'application/json',
+          'X-AppId': '0x0000000000000000000000000000000000000000000000000000000000000000',
+        },
+        method: 'GET',
+      }
+    )
+  }
+})
+
+test('Valid: Get Profile Data', async () => {
+  fetchMock.mockResponseOnce(JSON.stringify(PROFILE_DATA_RESPONSE), { status: HTTP_STATUS_OK })
+  const cowSdk1 = new CowSdk(chainId)
+  cowSdk1.updateChainId(1)
+
+  await cowSdk1.cowApi.getProfileData('0x6810e776880c02933d47db1b9fc05908e5386b96')
+  expect(fetchMock).toHaveBeenCalledTimes(1)
+  expect(fetchMock).toHaveBeenCalledWith(
+    'https://api.cow.fi/affiliate/api/v1/profile/0x6810e776880c02933d47db1b9fc05908e5386b96',
+    {
+      body: undefined,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-AppId': '0x0000000000000000000000000000000000000000000000000000000000000000',
+      },
+      method: 'GET',
+    }
+  )
+})
+
+test('Invalid: Get Profile Data from unexisting address', async () => {
+  fetchMock.mockResponseOnce(
+    JSON.stringify({
+      errorType: 'NotFound',
+      description: "You've passed an invalid URL",
+    }),
+    { status: HTTP_STATUS_NOT_FOUND }
+  )
+  const cowSdk1 = new CowSdk(chainId)
+  cowSdk1.updateChainId(1)
+
+  try {
+    await cowSdk1.cowApi.getProfileData('unexistingAddress')
+  } catch (e) {
+    const error = e as CowError
+    expect(error.message).toEqual("You've passed an invalid URL")
+    expect(fetchMock).toHaveBeenCalledTimes(1)
+    expect(fetchMock).toHaveBeenCalledWith('https://api.cow.fi/affiliate/api/v1/profile/unexistingAddress', {
+      body: undefined,
+      headers: {
+        'Content-Type': 'application/json',
+        'X-AppId': '0x0000000000000000000000000000000000000000000000000000000000000000',
+      },
+      method: 'GET',
+    })
+  }
+})
+
+test('Invalid: Get Profile Data from not supported network', async () => {
+  const profileData = await cowSdk.cowApi.getProfileData('0x6810e776880c02933d47db1b9fc05908e5386b96')
```

Comment:
Here it seems that you are simply getting a null response because the address is not found rather than querying on a different network as the test name suggests